### PR TITLE
Fix/identity tsql->databricks

### PR DESCRIPTION
**Pull request by dmoore247**
transpile from tsql to databricks was not generating the `ALWAYS` which is required by Databricks. 

>>> import sqlglot
>>> sqlglot.transpile('''CREATE TABLE dbo.Products (ProductID INT GENERATED AS IDENTITY (START WITH 1 INCREMENT BY 1) NOT NULL)''',read='tsql',write='databricks')
['CREATE TABLE dbo.Products (ProductID INT GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1) NOT NULL)']

### CODE REVIEW

File: sqlglot/dialects/databricks.py [hunk 1]
```diff
@@ -51,6 +51,15 @@ class Generator(Spark.Generator):
             exp.ToChar: lambda self, e: self.function_fallback_sql(e),
         }
 
+        PARAMETER_TOKEN = "$"
+
+        def generatedasidentitycolumnconstraint_sql(
+            self, expression: exp.GeneratedAsIdentityColumnConstraint
+        ) -> str:
+            expression = expression.copy()
+            expression.set("this", " ")  # trigger ALWAYS in super class
```
Comment (tobymao) [inline sqlglot/dialects/databricks.py:60]: it should be True

File: sqlglot/dialects/databricks.py [hunk 2]
```diff
@@ -51,6 +51,15 @@ class Generator(Spark.Generator):
             exp.ToChar: lambda self, e: self.function_fallback_sql(e),
         }
 
+        PARAMETER_TOKEN = "$"
```
Comment (tobymao) [inline sqlglot/dialects/databricks.py:54]: why did you add this back in? i think @GeorgeSittas removed it

Comment (dmoore247) [reply sqlglot/dialects/databricks.py:54]: I'll double check.

File: sqlglot/dialects/databricks.py (hunk 1 above)
Comment (dmoore247) [reply sqlglot/dialects/databricks.py:60]: Meaning set "this" to `True` ?

Comment (tobymao) [reply sqlglot/dialects/databricks.py:60]: yes

File: sqlglot/dialects/databricks.py (hunk 2 above)
Comment (dmoore247) [reply sqlglot/dialects/databricks.py]: George did remove it, my merge divereged so my mistake. Fixing...

**Comment by dmoore247** [comment]
Thanks!Sent via my iPhoneOn Sep 22, 2023, at 11:55 PM, Toby Mao ***@***.***> wrote:﻿
Merged #2300 into main.
